resources/js/components/Home/RegisterForm.vue

```diff
@@ -1,123 +1,73 @@
-        <div class="mt-2 w-full">
-          <input id="surname" v-model="form.surname" required name="surname" type="text"
-                 :class="{'ring-red focus:ring-red':errors.surname}" class="duration-200 ring-inset outline-none focus:ring focus:ring-primary bg-white/30 rounded-lg w-full p-3 text-gray-900 ring-2 ring-primary/30 placeholder:text-gray-400"
-          >
-          <div v-if="errors.surname" class="text-red">{{ errors.surname }}</div>
-        </div>
-      </div>
+      <CustomInput v-model="form.name" label="Imię" :error="errors.name" name="name" type="name" />
+      <CustomInput v-model="form.surname" label="Nazwisko" :error="errors.surname" name="surname" type="surname" />
     </div>
 
-    <div>
-      <div class="flex items-center justify-between">
-        <label for="school_id" class="block text-sm font-medium leading-6 text-gray-900">Szkoła</label>
-      </div>
-      <Searchbar :schools="schools" @change="school => form.school_id = school" />
-      <div v-if="errors.school_id" class="text-red">{{ errors.school_id }}</div>
-    </div>
+    <Searchbar
+      label="Szkoła"
+      aria-label="Poszukiwanie szkół. Otwiera listę szkół"
+      :options="filteredSchoolOptions"
+      :error="errors.school_id"
+      @change="school => form.school_id = school.id.toString()"
+    />
 
-    <div>
-      <label for="email" class="block text-sm font-medium leading-6 text-gray-900">E-mail</label>
-      <div class="mt-2">
-        <input id="email" v-model="form.email" required name="email" type="email" autocomplete="email"
-               :class="{'ring-red focus:ring-red':errors.email}" class="duration-200 ring-inset outline-none focus:ring focus:ring-primary bg-white/30 rounded-lg w-full p-3 text-gray-900 ring-2 ring-primary/30 placeholder:text-gray-400"
-        >
-        <div v-if="errors.email" class="text-red">{{ errors.email }}</div>
-      </div>
-    </div>
+    <CustomInput v-model="form.email" label="E-mail" :error="errors.email" name="email" type="email" />
 
-    <div>
-      <div class="flex items-center justify-between">
-        <label for="password" class="block text-sm font-medium leading-6 text-gray-900">Hasło</label>
-      </div>
-      <div class="mt-2">
-        <input id="password" v-model="form.password" required name="password" type="password" autocomplete="current-password"
-               :class="{'ring-red focus:ring-red':errors.password}" class="duration-200 ring-inset outline-none focus:ring focus:ring-primary bg-white/30 rounded-lg w-full p-3 text-gray-900 ring-2 ring-primary/30 placeholder:text-gray-400"
-        >
-        <div v-if="errors.password" class="text-red">{{ errors.password }}</div>
-      </div>
-    </div>
+    <PasswordInput v-model="form.password" :error="errors.password" />
 
-    <div class="mx-2 mt-4 flex flex-row items-center gap-4">
+    <label class="mx-2 mt-4 flex flex-row items-center gap-4">
```

Comment:
Label ? But where is `for` defended ?

Reply:
You don't need `for` if `label` wraps your input